Aplib.Tests/Belief/ExposedQueueTests.cs

```diff
@@ -0,0 +1,231 @@
+        ExposedQueue<int> queue = new(3);
+
+        // Act
+        int maxCount = queue.MaxCount;
+        int count = queue.Count;
+
+        // Assert
+        Assert.Equal(3, maxCount);
+        Assert.Equal(0, count);
+    }
+
+    [Fact]
+    public void ExposedQueue_WhenInitializedWithArray_ShouldHaveCorrectMaxCountAndCount()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new([1, 2, 3]);
+
+        // Act
+        int maxCount = queue.MaxCount;
+        int count = queue.Count;
+
+        // Assert
+        Assert.Equal(3, maxCount);
+        Assert.Equal(3, count);
+    }
+
+    [Fact]
+    public void ExposedQueue_WhenInitializedWithArrayAndCount_ShouldThrowExceptions()
+    {
+        // Arrange
+        void CountExceedsLength() => _ = new ExposedQueue<int>([1, 2, 3], 4);
+        void NegativeCount() => _ = new ExposedQueue<int>([1, 2, 3], -1);
+
+        // Assert
+        Assert.Throws<ArgumentOutOfRangeException>(CountExceedsLength);
+        Assert.Throws<ArgumentOutOfRangeException>(NegativeCount);
+    }
+
+    [Fact]
+    public void AccessIndex_WhenIndexIsOutOfBounds_ThrowsException()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new(3);
+        int elem;
+
+        // Act
+        void AccessNegativeIndex() => elem = queue[-1];
+        void AccessIndexGreaterThanCount() => elem = queue[queue.Count];
+
+        // Assert
+        Assert.Throws<ArgumentOutOfRangeException>(AccessNegativeIndex);
+        Assert.Throws<ArgumentOutOfRangeException>(AccessIndexGreaterThanCount);
+    }
+
+    [Fact]
+    public void Put_ArrayIsFull_WrapsAround()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new([4, 3, 2, 1]);
+
+        // Act
+        queue.Put(5);
+
+        // Assert
+        Assert.Equal(5, queue[0]);
+        Assert.Equal(4, queue[1]);
+        Assert.Equal(2, queue[3]);
+    }
+
+    [Fact]
+    public void GetHead_HeadIsUpdated_ReturnsCorrectElement()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new([3, 2, 1]);
+
+        // Act
+        queue.Put(4);
+
+        // Assert
+        Assert.Equal(2, queue.GetLast());
+    }
+
+    [Fact]
+    public void GetFirst_HeadIsUpdated_ReturnsFirstElement()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new([3, 2, 1]);
+
+        // Act
+        queue.Put(4);
+
+        // Assert
+        Assert.Equal(4, queue.GetFirst());
+    }
+
+    [Fact]
+    public void CopyTo_WrongArguments_ThrowsException()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new([3, 2, 1]);
+
+        // Act
+        void StartOutOfBounds() => queue.CopyTo(new int[3], -1, 2);
+        void EndOutOfOunds() => queue.CopyTo(new int[3], 0, 3);
+        void StartAfterEnd() => queue.CopyTo(new int[3], 2, 1);
+
+        // Assert
+        Assert.Throws<ArgumentOutOfRangeException>(StartOutOfBounds);
+        Assert.Throws<ArgumentOutOfRangeException>(EndOutOfOunds);
+        Assert.Throws<ArgumentException>(StartAfterEnd);
```

Comment:
Please write three separate tests for this. Asserting multiple things in a test is a test smell, as it is difficult to quickly pinpoint what the point of failure is.

In general, I'd think multiple asserts per test are allowed if you need to assert multiple thing with the same setup. This method uses two act methods and thus do not comply.